src/include/migraphx/op/pooling.hpp

```diff
@@ -89,6 +92,135 @@ struct pooling
         check_attribute_size();
         return stride.size();
     }
+
+    struct lpnorm_pool
+    {
+        int p = 0;
+
+        lpnorm_pool() = delete;
+
+        explicit lpnorm_pool(int x) : p{x} {};
+
+        template <class T>
+        double init()
+        {
+            return 0.0;
+        }
+
+        double operator()(double x, double y) const { return x + std::pow(std::abs(y), p); }
+
+        double final(double x, std::size_t) const { return std::pow(x, 1. / p); }
+    };
+
+    struct avg_pool
+    {
+
+        template <class T>
+        double init()
+        {
+            return 0.0;
+        }
+
+        double operator()(double x, double y) { return x + y; }
+
+        double final(double x, std::size_t y) { return (y == 0) ? 0.0 : (x / y); }
+    };
+
+    struct max_pool
+    {
+        template <class T>
+        T init()
+        {
+            return std::numeric_limits<T>::lowest();
+        }
+
+        double operator()(double x, double y) { return std::max(x, y); }
+
+        double final(double x, std::size_t) { return (x); }
+    };
+
+    template <typename TOut, typename TIn, typename Op>
```

Comment:
Dont prefix template parameter with `T`. We know its a template parameter because it uses CamelCase.

I also prefer to use `class` just because its shorter, but you dont have to change that if you dont like it.

Reply:
I will make the change to maintain consistency.